src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,242 @@
+            String command = words[0];
+            if (command.equals("list")) {
+                try {
+                    printList(tasks, words);
+                } catch (ExcessInputsException e) {
+                    System.out.println(LINE + "Too many inputs! Please only enter \"list\"\n" + LINE);
+                }
+            } else if (command.contains("mark")) {
+                try {
+                    editMarkStatus(tasks, words, command);
+                } catch (NumberFormatException e){
+                    System.out.println(LINE + "Mark/Unmark must only be followed by the index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(LINE + "Missing/Invalid task index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                } catch (NullPointerException e) {
+                    System.out.println(LINE + "Invalid task index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                }
+            } else if (command.equals("delete")) {
+                try {
+                    deleteTask(tasks, words);
+                } catch (NumberFormatException e){
+                    System.out.println(LINE + "Delete must only be followed by the index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println(LINE + "Missing/Invalid task index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                } catch (NullPointerException e) {
+                    System.out.println(LINE + "Invalid task index number!\n" +
+                            "Enter \"list\" to check the index.\n" + LINE);
+                }
+            } else {
+                try {
+                    switch(command) {
+                    case "todo":
+                        addTodo(tasks, words);
+                        break;
+                    case "deadline":
+                        addDeadline(tasks, words);
+                        break;
+                    case "event":
+                        addEvent(tasks, words);
+                        break;
+                    default :
+                        System.out.print(LINE + ERROR + LINE);
+                        line = in.nextLine();
+                        continue;
+                    }
+                    System.out.print(LINE + "    Got it. I've added this task:\n      " +
+                            tasks.get(tasks.size()-1).toString() + "\n    Now you have " + tasks.size() + " tasks in the list.\n" +
+                            LINE);
+                } catch (MissingDescriptionException e){
+                    System.out.println(LINE + "Missing task description details.\n" +
```

Comment:
consider refactoring this into the exception class since its the same message for all instances of this exception